### PR TITLE
Group kernels in dropdown when selecting in Config drawer

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Request from Chris to make our kernels dropdown look like this ("precisely this"):

kernels

### CODE REVIEW

File: packages/manager/src/cachedData/types.json [hunk 1]
```diff
@@ -1 +1 @@
...
...s":"dedicated","successor":null},{"id":"g1-gpu-rtx6000-1","label":"Dedicated 32GB + RTX6000 GPU x1","price":{"hourly":1.5,"monthly":1000},"addons":{"backups":{"price":{"hourly":0.06,"monthly":40}}},"memory":32768,"disk":655360,"transfer":16000,"vcpus":8,"gpus":1,"network_out":10000,"class":"gpu","successor":null},{"id":"g1-gpu-rtx6000-2","label":"Dedicated 64GB + RTX6000 GPU x2","price":{"hourly":3,"monthly":2000},"addons":{"backups":{"price":{"hourly":0.12,"monthly":80}}},"memory":65536,"disk":1310720,"transfer":20000,"vcpus":16,"gpus":2,"network_out":10000,"class":"gpu","successor":null},{"id":"g1-gpu-rtx6000-3","label":"Dedicated 96GB + RTX6000 GPU x3","price":{"hourly":4.5,"monthly":3000},"addons":{"backups":{"price":{"hourly":0.18,"monthly":120}}},"memory":98304,"disk":1966080,"transfer":20000,"vcpus":20,"gpus":3,"network_out":10000,"class":"gpu","successor":null},{"id":"g1-gpu-rtx6000-4","label":"Dedicated 128GB + RTX6000 GPU x4","price":{"hourly":6,"monthly":4000},"addons":{"backups":{"price":{"hourly":0.24,"monthly":160}}},"memory":131072,"disk":2621440,"transfer":20000,"vcpus":24,"gpus":4,"network_out":10000,"class":"gpu","successor":null}],"page":1,"pages":1,"results":28}
```
Comment (Jskobos) [inline packages/manager/src/cachedData/types.json:1]: I ran `yarn build` to cache the kernels, this got updated as a side effect

**Review by tiffwong — APPROVED** [review]
Looks good!

File: packages/api-v4/src/linodes/types.ts [hunk 2]
```diff
@@ -186,14 +186,17 @@ export interface Devices {
...
 export interface Kernel {
   id: string;
   label: string;
   version: string;
   kvm: boolean;
   xen: boolean;
-  architecture: string;
+  architecture: KernelArchitecture;
   pvops: boolean;
+  deprecated: boolean;
```
Comment (DevDW) [inline packages/api-v4/src/linodes/types.ts:199]: The screenshot says "depreciated," but I agree that it should be "deprecated"

**Review by DevDW — APPROVED** [review]

**Review by johnwcallahan — CHANGES_REQUESTED** [review]
Great work with the sorting and grouping: clean and readable as always. Left one comment about a failure scenario.

**Review by johnwcallahan — APPROVED** [review]

**Pull request merged**